File: remove_dark_mode.py
import os
import re

def main():
    root_dir = '/home/deploy/bahrawy'
    target_exts = ('.ts', '.tsx')
    changed_files = 0

    pattern = re.compile(r'(?:\s*)dark:[a-zA-Z0-9\-\/\[\]#\.]+(?:\s*)')

    for dirpath, dirnames, filenames in os.walk(root_dir):
        if 'node_modules' in dirpath or '.next' in dirpath:
            continue
        for filename in filenames:
            if filename.endswith(target_exts):
                filepath = os.path.join(dirpath, filename)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()

                new_content = pattern.sub(' ', content)

                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    changed_files += 1

    print(f"Removed dark mode variants in {changed_files} files.")

if __name__ == '__main__':
    main()

File: remove_dark_mode.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// Find all tsx and ts files
const files = execSync('find apps packages -type f -name "*.tsx" -o -name "*.ts"', { encoding: 'utf-8' })
  .split('\n')
  .filter(Boolean);

let changedFiles = 0;

for (const file of files) {
  let content = fs.readFileSync(file, 'utf-8');
  
  // This regex matches "dark:" followed by any tailwind class characters
  // It handles spacing gracefully
  const newContent = content.replace(/\bdark:[a-zA-Z0-9\-\/\[\]#\.]+\b/g, (match) => {
    return '';
  }).replace(/ +/g, ' '); // Clean up extra spaces temporarily, but wait this might mess up formatting
  
  // A safer regex replacement:
  // Replace "dark:something" and a potential trailing or leading space.
  let saferContent = content.replace(/(?:\s*)dark:[a-zA-Z0-9\-\/\[\]#\.]+(?:\s*)/g, ' ');
  
  if (content !== saferContent) {
    fs.writeFileSync(file, saferContent, 'utf-8');
    changedFiles++;
  }
}

console.log(`Removed dark mode variants in ${changedFiles} files.`);
